src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs

```diff
@@ -326,6 +326,12 @@ private void VerifyType(DataContract dataContract, Type declaredType)
                 scopedKnownTypes.Pop();
             }
         }
+
+        internal static void WriteJsonNameWithMapping(XmlWriterDelegator xmlWriter, XmlDictionaryString[] memberNames, int index)
```

Comment:
It's accessed via reflection (I had to search for it, as I had a similar question in mind); see my comment for a link to the access site.